### PR TITLE
Adding inputmode option to input component

### PR DESCRIPTION
**Pull request by htmlandbacon**
This adds the option of setting an optional [inputmode](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode).

One thing I couldn't see was an example of a check against something that isn't there by default?

I.e if inputmode="{{ params.inputmode }}" wasn't wrapped correctly it would always be there.

Just just added a check, happy to remove, or change if required.

I figured this was an easy step towards improving https://github.com/alphagov/govuk-frontend/issues/1449#issuecomment-504006087

### CODE REVIEW

File: src/govuk/components/input/template.njk [hunk 1]
```diff
@@ -42,5 +42,6 @@
   {%- if describedBy %} aria-describedby="{{ describedBy }}"{% endif %}
   {%- if params.autocomplete %} autocomplete="{{ params.autocomplete}}"{% endif %}
   {%- if params.pattern %} pattern="{{ params.pattern }}"{% endif %}
+  {%- if params.inputmode %} inputmode="{{ params.inputmode }}" {% endif %}
```
[the remark this answers is not in the document]
Comment (htmlandbacon) [reply src/govuk/components/input/template.njk]: Good spot, have updated 👍

File: src/govuk/components/input/template.test.js [hunk 2]
```diff
@@ -112,6 +112,13 @@ describe('Input', () => {
       const $formGroup = $('.govuk-form-group')
       expect($formGroup.hasClass('extra-class')).toBeTruthy()
     })
+
+    it('renders without inputmode', () => {
```
[the remark this answers is not in the document]
Comment (htmlandbacon) [reply src/govuk/components/input/template.test.js]: I've removed this test.

I guess its always that worry that someone introduces something weird - but negative testing always seems a bit janky.

**Comment by htmlandbacon** [comment]
Updated with the comments in mind 👍

**Comment by nickcolley** [comment]
Thanks Colin :)

**Pull request merged**